python/Ganga/GPIDev/Lib/Job/Job.py

```diff
@@ -2181,15 +2183,31 @@ def _repr(self):
 # return tuple(index)
 
     def _subjobs_proxy(self):
+
         from Ganga.GPIDev.Lib.Registry.JobRegistry import JobRegistrySlice, _wrap
-        subjobs = JobRegistrySlice('jobs(%s).subjobs' % str(self.id))
-        if isType(self.subjobs, SubJobXMLList):
-            subjobs.objects = self.subjobs
-        else:
-            for sj in self.subjobs:
-                subjobs.objects[sj.id] = sj
-        #print('return slice: %s' % str(subjobs))
-        return _wrap(subjobs)
+        from Ganga.GPIDev.Base.Proxy import stripProxy
+        if not isType(self._stored_subjobs_proxy, JobRegistrySlice):
+            subjob_slice = JobRegistrySlice('jobs(%s).subjobs' % str(self.id))
+            self._stored_subjobs_proxy = _wrap(subjob_slice)
+
+        if len(self._stored_subjobs_proxy) != len(self.subjobs):
+
+            if isType(self.subjobs, SubJobXMLList):
+                subjob_slice.objects = self.subjobs
+                #self._stored_subjobs_proxy = _wrap(stripProxy(self._stored_subjobs_proxy))
+            elif isType(self.subjobs, (list, GangaList)):
+                subjob_slice = stripProxy(self._stored_subjobs_proxy)
+                if len(subjob_slice.objects.keys()):
```

Comment:
`if subjob_slice.objects:` will use `__len__` ([see this](https://docs.python.org/2/reference/datamodel.html#object.__nonzero__)) which `SubJobXMLList` implements correctly so it should indeed work.